types/consensus_data.go

```diff
@@ -112,23 +111,51 @@ type ConsensusData struct {
 	//			13*SignedPartialSignatureMessage(2^16) ~= 2^20
 	PreConsensusJustifications []*SignedPartialSignatureMessage `ssz-max:"13"`
 	// DataSSZ has max size as following
-	// Biggest object is a full beacon block
-	// BeaconBlock is 2*32+2*8 + BeaconBlockBody
+	// Biggest object is a Deneb.BlockContents with:
+	// - KZGProofs: 6 * 48 = 288
+	// - Blobs: 6 * 131072 = 786432
+	// - A BeaconBlock: 2*32+2*8 + BeaconBlockBody
 	// BeaconBlockBody is
 	//			96 + ETH1Data(2*32+8) + 32 +
 	//			16*ProposerSlashing(2*SignedBeaconBlockHeader(96 + 3*32 + 2*8)) +
-	//			2*AttesterSlashing(2*IndexedAttestation(2048*8 + 96 + AttestationData(2*8 + 32 + 2*(8+32)))) +
-	//			128*Attestation(2048*8 + 96 + AttestationData(2*8 + 32 + 2*(8+32))) +
+	//			2*AttesterSlashing(2*IndexedAttestation(2048 + 96 + AttestationData(2*8 + 32 + 2*(8+32)))) +
+	//			128*Attestation(2048 + 96 + AttestationData(2*8 + 32 + 2*(8+32))) +
 	//			16*Deposit(33*32 + 48 + 32 + 8 + 96) +
 	//			16*SignedVoluntaryExit(96 + 2*8) +
 	//			SyncAggregate(64 + 96) +
-	//			ExecutionPayload(32 + 20 + 2*32 + 256 + 32 + 4*8 + 3*32 + 1048576*1073741824)
-	// = 2^21(everything but transactions) + 2^50 (transaction list)
+	//			ExecutionPayload(32 + 20 + 2*32 + 256 + 32 + 4*8 + 3*32 + 1048576*1073741824 + 16 * (2*8 + 20 + 8) + 8 + 8) +
+	//			BLSToExecutionChanges 16 * (96 + (8 + 48 + 20)) +
+	//			KZGCommitment 4096 * 48
+	// = 1315964 (everything but transactions) + 2^50 (transaction list)
 	// We do not need to support such a big DataSSZ size as 2^50 represents 1000X the actual block gas limit
 	// Current 30M gas limit produces 30M / 16 (call data cost) = 1,875,000 bytes (https://eips.ethereum.org/EIPS/eip-4488)
-	// we can upper limit transactions to 2^21, together with the rest of the data 2*2^21 = 2^22 = 4,194,304 bytes
-	// Exmplanation on why transaction sizes are so big https://github.com/ethereum/consensus-specs/pull/2686
-	DataSSZ []byte `ssz-max:"4194304"` // 2^22
+	// we can upper limit transactions to 2^21, together with the rest of the data 1315964 + 2^21  = 3,413,116 bytes
```

Comment:
Why did we go down from 2*2^21 to 2^21? If anything, blob size should have increased our ssz-max, no?

Reply:
See this comment where we corrected a previous mistake regarding attestations:
https://github.com/bloxapp/ssv-spec/pull/324#discussion_r1455227526

checkout the commented python script

Reply:
> they're thinking of increasing gas size from 30M to 40M very soon
When they are done thinking we'll fix this

it will only be on the next hard fork (best case) so there is time

Reply:
> > they're thinking of increasing gas size from 30M to 40M very soon
> > When they are done thinking we'll fix this
> 
> it will only be on the next hard fork (best case) so there is time

I think maybe we'd like to support a probable gas increase without having to fork?